xcdat/temporal.py

```diff
@@ -669,15 +671,38 @@ def departures(
             reference_period,
             season_config,
         )
-        ds_climo = ds_climo.rename({self.dim: self._labeled_time.name})
+
+        # 4. Group the averaged data variable values by the time `freq`.
+        # ----------------------------------------------------------------------
+        # This step allows us to perform xarray's grouped arithmetic to
+        # calculate departures.
+        dv_obs = ds_avg[data_var].copy()
+        self._labeled_time = self._label_time_coords(dv_obs[self.dim])
+        dv_obs_grouped = self._group_data(dv_obs)
+
+        # 5. Align time dimension names using the labeled time dimension name.
+        # ----------------------------------------------------------------------
+        # The climatology's time dimension is renamed to the labeled time
+        # dimension in step #4 above (e.g., "time" -> "season"). xarray requires
+        # dimension names to be aligned to perform grouped arithmetic, which we
+        # use for calculating departures in step #5. Otherwise, this error is
+        # raised: "`ValueError: incompatible dimensions for a grouped binary
+        # operation: the group variable '<FREQ ARG>' is not a dimension on the
+        # other argument`".
         dv_climo = ds_climo[data_var]
+        dv_climo = dv_climo.rename({self.dim: self._labeled_time.name})
 
-        # 4. Calculate the departures for the data variable using the formula,
-        # grouped observation data - climatology.
+        # 6. Calculate the departures for the data variable.
+        # ----------------------------------------------------------------------
+        # departures = observation - climatology
         with xr.set_options(keep_attrs=True):
             dv_departs = dv_obs_grouped - dv_climo
             dv_departs = self._add_operation_attrs(dv_departs)
             ds_avg[data_var] = dv_departs
+
+            # The original time dimension name is restored after grouped
+            # arithmetic, so the labeled time dimension name is no longer needed
+            # and therefore dropped.
             ds_avg = ds_avg.drop_vars(self._labeled_time.name)
 
         if weighted and keep_weights:
```

Comment:
In `departures()`, I updated the order of operations and added clearer comments to outline the steps and why they need to be done this specific way.